modules/mixcr/1.2/mixcr.smk

```diff
@@ -0,0 +1,320 @@
+    _mixcr_to_fasta,
+    _igblastn_run,
+    _update_mixcr_results,
+    _symlink_mixcr_update,
+    _mixcr_chains,
+    _mixcr_all,
+
+
+##### RULES #####
+
+
+# Symlinks the input files into the module results directory (under '00-inputs/')
+rule _mixcr_input_fastq:
+    input:
+        fastq_1 = CFG["inputs"]["sample_fastq_1"],
+        fastq_2 = CFG["inputs"]["sample_fastq_2"],
+    output:
+        fastq_1 = CFG["dirs"]["inputs"] + "fastq/{seq_type}/{sample_id}.R1.fastq.gz",
+        fastq_2 = CFG["dirs"]["inputs"] + "fastq/{seq_type}/{sample_id}.R2.fastq.gz",
+    run:
+        op.absolute_symlink(input.fastq_1, output.fastq_1)
+        op.absolute_symlink(input.fastq_2, output.fastq_2)
+
+# Installs latest MiXCR release from github if the mixcr folder is not present yet
+rule _install_mixcr:
+    params:
+        mixcr = CFG["inputs"]["mixcr_exec"]
+    output: 
+        complete = CFG["inputs"]["mixcr_exec"] + "/mixcr_dependencies_installed.success"
+    shell:
+        '''
+        download_url=$(curl --silent "https://api.github.com/repos/milaboratory/mixcr/releases/latest" | grep '"browser_download_url":' | sed -E 's/.*\"([^\"]+)\".*/\\1/');
+        mkdir -p {params.mixcr};
+
+        if [ ! -f {params.mixcr}/mixcr ]; then
+            wget -cO - $download_url > {params.mixcr}/mixcr.zip && unzip {params.mixcr}/mixcr.zip -d {params.mixcr}/ && rm {params.mixcr}/mixcr.zip;
+            mv {params.mixcr}/mixcr*/* {params.mixcr}/ && rm -r {params.mixcr}/mixcr*/;
+        fi
+
+        touch  {output.complete};
+        '''
+
+# Run MiXCR rule
+rule _mixcr_run:
+    input:
+        fastq_1 = rules._mixcr_input_fastq.output.fastq_1,
+        fastq_2 = rules._mixcr_input_fastq.output.fastq_2,
+        fastq_1_real = CFG["inputs"]["sample_fastq_1"], # Prevent premature deletion of fastqs marked as temp
+        fastq_2_real = CFG["inputs"]["sample_fastq_2"],
+        installed = rules._install_mixcr.output.complete
+    output:
+        txt = CFG["dirs"]["mixcr"] + "{seq_type}/{sample_id}/mixcr.{sample_id}.clonotypes.ALL.txt",
+        report = CFG["dirs"]["mixcr"] + "{seq_type}/{sample_id}/mixcr.{sample_id}.report",
+        results = expand(CFG["dirs"]["mixcr"] + "{{seq_type}}/{{sample_id}}/mixcr.{{sample_id}}.clonotypes.{chain}.txt", chain = RECEPTORS)
+    log:
+        stdout = CFG["logs"]["mixcr"] + "{seq_type}/{sample_id}/mixcr_run.stdout.log",
+        stderr = CFG["logs"]["mixcr"] + "{seq_type}/{sample_id}/mixcr_run.stderr.log"
+    resources:
+        **CFG["resources"]["mixcr_run"]
+    params:
+        opts = op.switch_on_wildcard("seq_type", CFG["options"]["mixcr_run"]),
+        prefix = CFG["dirs"]["mixcr"] + "{seq_type}/{sample_id}/mixcr.{sample_id}", 
+        mixcr = CFG["inputs"]["mixcr_exec"] + "/mixcr", 
+        jvmheap = lambda wildcards, resources: int(resources.mem_mb * 0.8) 
+    conda: CFG["conda_envs"]["java"]
+    threads:
+        CFG["threads"]["mixcr_run"]
+    message:
+        "{params.mixcr}"    
```

Comment:
Can you make this more informative, e.g. `Using MiXCR installation at {params.mixcr}.` Or delete it, I'm not sure it's necessary.

Reply:
I don't think it's necessary either, I'll delete it